pkg/placement/membership.go

```diff
@@ -273,14 +275,18 @@ func (p *Service) performTableDissemination() {
 		log.Infof(
 			"Start disseminating tables. memberUpdateCount: %d, streams: %d, targets: %d, table generation: %s",
 			cnt, nStreamConnPool, nTargetConns, state.Version)
-		p.performTablesUpdate(p.streamConnPool, state)
+		p.streamConnPoolLock.RLock()
+		streamConnPool := make([]placementGRPCStream, len(p.streamConnPool))
+		copy(streamConnPool, p.streamConnPool)
+		p.streamConnPoolLock.RUnlock()
```

Comment:
Instead of locking again here, hold the lock from the first line and defer the Runlock(). It would also make sure the pool has not changed during the whole method execution.

Reply:
`performTablesUpdate` may be time-consuming which contains lots of network writes. If we hold the `streamConnPoolLock` lock during whole `performTableDissemination` execution, it would block `addStreamConn` and `deleteStreamConn` with long time. Does it matter?

> Instead of locking again here, hold the lock from the first line and defer the Runlock(). It would also make sure the pool has not changed during the whole method execution.

`performTablesUpdate` may be time-consuming which contains lots of network writes. If we hold `streamConnPoolLock` lock during whole `performTableDissemination` execution, it would block `addStreamConn` and `deleteStreamConn` with long time. Does it matter?

@artursouza

Reply:
This is about correctness of the state and not about performance. We should revisit this because I think we want to guarantee the locked value does not change between the two locks.